mapping.go

```diff
@@ -129,9 +146,31 @@ func (o *mapAssignmentKind) String() string {
 	if o.ElemDirect {
 		s += " (direct)"
 	}
+	if o.ElemConvert {
+		s += " (convert)"
+	}
 	return s
 }
 
+func convertibleButNotIdentical(typ, typeDecode types.Type) bool {
+	// Only allow this for basic and named.
+
+	switch typeDecode.(type) {
+	case *types.Basic, *types.Named:
+	default:
+		return false
+	}
+
+	if types.Identical(typ, typeDecode) {
```

Comment:
for example `typ=string` will have `typeDecode=string` too and `string` is convertible to `string` but we don't want that to trigger the special code.